hermit-sys/src/net/mod.rs

```diff
@@ -357,9 +347,9 @@ pub(crate) fn network_delay(timestamp: Instant) -> Option<Duration> {
 }
 
 pub(crate) async fn network_run() {
-	future::poll_fn(|cx| match NIC.lock().deref_mut() {
+	future::poll_fn(|_cx| match NIC.lock().deref_mut() {
```

Comment:
Never waking the waker might cause problems. We might have to rethink how this function should work in an async context.

Reply:
Smoltcp is waking the waker.

Reply:
But smoltcp is never getting the waker from `_cx` like this.

Reply:
We register the waker i.e. at https://github.com/hermitcore/rusty-hermit/blob/master/hermit-sys/src/net/mod.rs#L290

Reply:
But this `async fn` does not register its waker.

* We could wake the waker immediately, signaling that we want to poll again immediately, but I am not sure if that makes sense for us.
* We could supply the waker to smoltcp, but I am not sure if that is possible.
* We could make this a non-`async` function, as its inner operations are not async as well.

Reply:
It is a background task. It runs only if blocked_on has time. It is called by run_executor.

Reply:
Ah, I understand. The problem was [`fd23cdd#R31-R42`](https://github.com/hermitcore/rusty-hermit/commit/fd23cddf2d8027ee0fca7396ad5cc03549b283e3#diff-9a7af76af887c12192106d6e158ec152c277af56adcb4a96c6d8bd86755fdb07R31-R42).

That change should be reverted. All futures should take care of waking themselves. I'll create a PR. After that, this function should always wake itself immediately.

Reply:
I reverted already in this PR

Reply:
Where? You can incorporate https://github.com/hermitcore/rusty-hermit/pull/346

Reply:
Ah, now I understand and revise the PR. Does it fulfill your requirements?

Reply:
Yes, thank you! :D